packages/cli-kit/src/public/node/fs.ts

```diff
@@ -468,8 +468,16 @@
 
   /** Type of word to use for random name. */
   family?: RandomNameFamily
+
+  /** Used to to track how many attempts have been made to generate the name. */
+  attempts?: number
+
+  /** Predicate for validating the generated name. */
+  isValidName?: (name: string) => boolean
```

Comment:
i'm not sure about this, the function should be simple, just generate a name following some options (suffix, family and the directory).

It should be responsibility of the caller to validate the proposed name and decide whether or not to try again (and how many times).

`attemps` should definitely not be part of the options if you want to use it to track the recursion, because it can be set externally and any caller will see this and interpret it as "max attemps" and set a number.

My proposal: 
- keep `generateRandomNameForSubdirectory` as it is now
- Make the app init validate the result name and retry again if needed
- Remove the extra long word, just to be sure.

What do you think?

Reply:
Sounds good to me! Thanks, Isaac!